src/main/scala/io/applicative/datastore/DatastoreService.scala

```diff
@@ -0,0 +1,188 @@
+package io.applicative.datastore
+import javax.inject.Singleton
+
+import com.google.cloud.datastore.{DatastoreOptions, DatastoreReader, Entity, KeyFactory, Transaction, Datastore => CloudDataStore, Key => CloudKey}
+import io.applicative.datastore.exception.UnsupportedIdTypeException
+import io.applicative.datastore.util.reflection.ReflectionHelper
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.reflect.runtime.universe._
+
+@Singleton
```

Comment:
Any advantage to annotate the class with `@Singleton`, versus sticking all the functionality into `object`?